integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/RddPathUtils.java

```diff
@@ -115,11 +115,14 @@ public boolean isDefinedAt(Object rdd) {
 
     @Override
     public Stream<Path> extract(ParallelCollectionRDD rdd) {
+      int SEQ_LIMIT = 1000;
       try {
         Object data = FieldUtils.readField(rdd, "data", true);
         log.debug("ParallelCollectionRDD data: {}", data);
-        if (data instanceof Seq) {
-          return ScalaConversionUtils.fromSeq((Seq) data).stream()
+        if ((data instanceof Seq) && ((Seq) data).head() instanceof Tuple2) {
+          // exit if the first element is invalid
+          Seq data_slice = (Seq) ((Seq) data).slice(0, SEQ_LIMIT);
+          return ScalaConversionUtils.fromSeq(data_slice).stream()
```

Comment:
Right. Should we have a boolean flag, and perhaps log the error just once?
In that case, what should we do about `el.getClass().getCanonicalName()` then, in case different elements belong to different classes?

Reply:
I have added this for now; it will log the first element that does not match the predicate.
Please let me know whether this is okay.